src/Fixer/ControlStructure/NoBreakCommentFixer.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\ControlStructure;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Fixer\WhitespacesAwareFixerInterface;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * Fixer for rule defined in PSR2 ¶5.2.
+ */
+final class NoBreakCommentFixer extends AbstractFixer implements WhitespacesAwareFixerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'There must be a comment when fall-through is intentional in a non-empty case body.',
+            array(new CodeSample("<?php\nswitch (\$foo)\n{    case 1:\n        foo();\n    case 2:\n        bar();\n}"))
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(T_CASE);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach ($tokens as $position => $token) {
+            if ($token->isGivenKind(T_CASE)) {
+                $this->fixCase($tokens, $position);
+            }
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $casePosition
+     */
+    private function fixCase(Tokens $tokens, $casePosition)
+    {
+        $empty = true;
+        $commentPosition = null;
+        for ($i = $tokens->getNextTokenOfKind($casePosition, array(':', ';')) + 1, $max = count($tokens); $i < $max; ++$i) {
+            if ($tokens[$i]->isGivenKind(array(T_SWITCH, T_IF, T_ELSE, T_ELSEIF, T_FOR, T_FOREACH, T_WHILE, T_DO, T_FUNCTION))) {
+                $empty = false;
+                $i = $this->getStructureEnd($tokens, $i);
+
+                continue;
+            }
+
+            if ($tokens[$i]->isGivenKind(array(T_BREAK, T_CONTINUE, T_RETURN))) {
+                if (null !== $commentPosition) {
+                    $this->removeComment($tokens, $commentPosition);
+                }
+
+                break;
+            }
+
+            if ($tokens[$i]->isGivenKind(array(T_COMMENT)) && '// no break' === $tokens[$i]->getContent()) {
+                $commentPosition = $i;
+                continue;
+            }
+
+            if ($tokens[$i]->isGivenKind(array(T_CASE, T_DEFAULT))) {
+                if (!$empty) {
```

Comment:
It would require to write:

```php
if ($empty) {
    if (null !== $commentPosition) {
        $this->removeComment($tokens, $commentPosition);
    }
} elseif (null === $commentPosition) {
    $this->insertCommentAt($tokens, $i);
} else {
    $this->ensureNewLineAt($tokens, $commentPosition);
}
```
and having a single `if` inside another `if` looks weird to me.